python/cugraph/components/cc_wrapper.pyx

```diff
@@ -0,0 +1,61 @@
+# Copyright (c) 2019, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from c_cc cimport *
+#from libcpp cimport bool
+from c_graph cimport *
+#from libcpp cimport cugraph_cc_t
+from libc.stdint cimport uintptr_t
+import cudf
+#from pygdf import Column
+import numpy as np
+
+cpdef weak_cc(G, connect_type = CUGRAPH_WEAK):
+    """
+    Generate the connected components and attach a component label to each vertex.
+    
+    Parameters
+    ----------
+    G : cugraph.graph
+        cuGraph graph descriptor, should contain the connectivity information as an
+        adjacency list.
+    connect_type : cugraph_cc_t
+        Weak (CUGRAPH_WEAK), or Strong (CUGRAPH_STRONG) (not implemented, yet)
+    
+    Returns
+    -------
+    df : cudf.DataFrame
+        df['labels'][i] gives the label id of the i'th vertex
```

Comment:
Nx generate a sorted list of connected components, largest first. We should do the same.

Reply:
Why? WeaklyCC algorithm returns an array of labels for efficiency. Creating an array of sets takes away from performance.

Reply:
Also, these requests must be made before the API is created, not after. We had lengthy conversations about the WeakCC API. Changing it at the point of PR aproval is _not_ the right time to do it.

Reply:
User may expect cuGraph to return the result in the same way that Nx does.
I don't think returning them sorted makes a big performance difference.

@BradReesWork please confirm if we should match Nx API or if you are fine with having this mismatch.

Reply:
> Also, these requests must be made before the API is created, not after. We had lengthy conversations about the WeakCC API.

The general guideline is to provide at least one API that is as close as Nx as possible.
It is actually the first time I see the python AP for WCC, I don't recall seeing an earlier API proposal for python.

Reply:
Please wait for @BradReesWork to resolve this

Reply:
What is the API proposal procedure? Are there API proposals for the other features? Can you point me to them? Because as far as WeakCC is concerned I believe we discussed it extensively in the meetings, have we not? So, then what exactly is missing from so that you can "see" an API proposal?

Reply:
Andrei, I don't think we need a formal API proposal. 
I am just pointing out that Nx returns a different thing while the guideline is to match Nx API. I don't see a strong motivation on why we cannot sort it as Nx does but perhaps I am missing something. Let's wait on Brad's' input here as well.

Reply:
I do not think that we should worry about sorting the output, or making list.  Returning a DataFrame is enough.  the user can easily do df.sort_values(...).
also a dataframe allows the results to be merged (joined) back into the intiial data set and additional processing run